cpmpy/solvers/cpo.py

```diff
@@ -0,0 +1,491 @@
+        elif self.cpo_status == "Unknown":
+            # can happen when timeout is reached...
+            self.cpm_status.exitstatus = ExitStatus.UNKNOWN
+        elif self.cpo_status == "Optimal":
+            self.cpm_status.exitstatus = ExitStatus.OPTIMAL
+        elif self.cpo_status == "JobFailed":
+            self.cpm_status.exitstatus = ExitStatus.ERROR
+        elif self.cpo_status == "JobAborted":
+            self.cpm_status.exitstatus = ExitStatus.NOT_RUN
+        else:  # another?
+            raise NotImplementedError(self.cpo_status)  # a new status type was introduced, please report on GitHub
+
+        # True/False depending on self.cpm_status
+        has_sol = self._solve_return(self.cpm_status)
+
+        # translate solution values (of user specified variables only)
+        if has_sol:
+            # fill in variable values
+            for cpm_var in self.user_vars:
+                sol_var = self.solver_var(cpm_var)
+                if isinstance(cpm_var,_BoolVarImpl):
+                    # because cp optimizer does not have boolean variables we use an integer var x with domain 0, 1
+                    # and then replace a boolvar by x == 1
+                    sol_var = sol_var.children[0]
+                    cpm_var._value = bool(self.cpo_result.get_var_solution(sol_var).get_value())
+                else:
+                    cpm_var._value = self.cpo_result.get_var_solution(sol_var).get_value()
+
+            # translate objective, for optimisation problems only
+            if self.has_objective():
+                self.objective_value_ = self.cpo_result.get_objective_value()
+
+        else:  # clear values of variables
+            for cpm_var in self.user_vars:
+                cpm_var.clear()
+
+        return has_sol
+
+    def solveAll(self, display=None, time_limit=None, solution_limit=None, call_from_model=False, **kwargs):
+        """
+            A shorthand to (efficiently) compute all (optimal) solutions, map them to CPMpy and optionally display the solutions.
+
+            If the problem is an optimization problem, returns only optimal solutions.
+
+           Arguments:
+                - display: either a list of CPMpy expressions, OR a callback function, called with the variables after value-mapping
+                        default/None: nothing displayed
+                - time_limit: stop after this many seconds (default: None)
+                - solution_limit: stop after this many solutions (default: None)
+                - call_from_model: whether the method is called from a CPMpy Model instance or not
+                - any other keyword argument
+
+            Returns: number of solutions found
+        """
+
+        import docplex.cp as docp
+        #cpo_solver = self.cpo_model.start_search(TimeLimit=time_limit, SolutionLimit=solution_limit, LogVerbosity='Quiet', **kwargs)
+        solution_count = 0
+        if solution_limit is None:
+            solution_limit = 99999
```

Comment:
that is not advisable... we can solve nqueens over 100k in a very fast way.
Change the 'while' below to 'while solution_limit is None or solution_count < lmt'?